alea/src/covariance.cpp

```diff
@@ -190,23 +212,66 @@ template class cov_result<std::complex<double>, elliptic_var>;
 
 
 template <typename T, typename Str>
-void serialize(serializer &s, const cov_result<T,Str> &self)
+void serialize(serializer &s, const std::string &key, const cov_result<T,Str> &self)
 {
     internal::check_valid(self);
-    s.write("count", make_adapter(self.count()));
-    s.write("obs", make_adapter(self.observations()));
-    s.write("mean/value", make_adapter(self.mean()));
-    s.write("mean/error", make_adapter(self.stderror()));
+    internal::serializer_sentry group(s, key);
+
+    serialize(s, "@size", self.store_->data_.size());
+    serialize(s, "count", self.store_->count_);
+    serialize(s, "count2", self.store_->count2_);
+    s.enter("mean");
+    serialize(s, "value", self.store_->data_);
+    serialize(s, "error", self.stderror());   // TODO temporary
```

Comment:
I understand that in general numbers of calls to `serialize()` and `deserialize()` don't have to match.
However, how am I going to implement `stream_(de)serializer` (#446)?
All it would be able to do is to write/read the next value to/from stream ignoring key names altogether.

Reply:
> I understand that in general numbers of calls to serialize() and deserialize() don't have to match.
However, how am I going to implement stream_(de)serializer (#446)?
All it would be able to do is to write/read the next value to/from stream ignoring key names altogether.

Very good point. Please look at my fix for this.